Source/Core/DolphinQt/GameList/GameList.cpp

```diff
@@ -576,8 +577,19 @@ void GameList::OpenContainingFolder()
   if (!game)
     return;
 
-  QUrl url = QUrl::fromLocalFile(
-      QFileInfo(QString::fromStdString(game->GetFilePath())).dir().absolutePath());
+  // Remove everything after the last separator in the game's path, resulting in the parent
+  // directory path with a trailing separator. Keeping the trailing separator prevents Windows from
+  // mistakenly opening a .bat or .exe file in the grandparent folder when that file has the same
+  // base name as the folder (See https://bugs.dolphin-emu.org/issues/12411).
+  const std::string game_path = game->GetFilePath();
+  const size_t last_separator_index = game_path.rfind(DIR_SEP_CHR);
```

Comment:
It's probably best to use `SplitPath` (from StringUtil.h) here for clarity reasons.

Reply:
Thank you, I looked for a function to get the directory but didn't find that one.